development/cpp/internal_rendering_architecture.rst

```diff
@@ -0,0 +1,600 @@
+.. _doc_internal_rendering_architecture:
+
+Internal rendering architecture
+===============================
+
+This page is a high-level overview of Godot 4's internal renderer design.
+It does not apply to previous Godot versions.
+
+The goal of this page is to document design decisions taken to best suit
+:ref:`Godot's design philosophy <doc_best_practices_for_engine_contributors>`,
+while providing a starting point for new rendering contributors.
+
+If you have questions about rendering internals not answered here, feel free to
+ask in the ``#rendering`` channel of the
+`Godot Contributors Chat <https://chat.godotengine.org/channel/rendering>`__.
+
+Rendering drivers
+-----------------
+
+Godot 4 supports the following graphics APIs:
+
+Vulkan
+^^^^^^
+
+This is the main driver in Godot 4, with most of the development focus going
+towards this driver.
+
+Vulkan 1.0 is required as a baseline, with optional Vulkan 1.1 and 1.2 features
+used when available. `volk <https://github.com/zeux/volk>`__ is used as a Vulkan
+loader, and
+`Vulkan Memory Allocator <https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator>`__
+is used for memory management.
+
+Both the Clustered Forward and Forward Mobile
+:ref:`doc_internal_rendering_architecture_backends` are supported when using the
+Vulkan driver.
+
+**Vulkan context creation:**
+
+- `drivers/vulkan/vulkan_context.cpp <https://github.com/godotengine/godot/blob/master/drivers/vulkan/vulkan_context.cpp>`__
+
+Direct3D 12
+^^^^^^^^^^^
+
+Like Vulkan, the Direct3D 12 driver targets modern platforms only.
+**TODO:** Mention required feature level/optional feature levels.
```

Comment:
@RandomShaper What's the minimum required D3D12 feature level for the renderer, and what's the optimal level (i.e. with all optional features covered)?

Knowing this would also be useful to list supported GPUs on a troubleshooting page. Specifically, there are some AMD GPUs that support Direct3D 12, but only with a limited feature level (Radeon 200/300 series).